core/store/models/service_agreement.go

```diff
@@ -192,15 +198,12 @@ func (e Encumbrance) ABI() ([]byte, error) {
 	}
 
 	// Absolute end date as a big-endian uint32 (unix seconds)
-	endAt := e.EndAt.Time.Unix()
-	if endAt > 0xffffffff { // Optimistically, this could be an issue in 2038...
-		return nil, fmt.Errorf(
-			"endat date %s is too late to fit in uint32",
-			e.EndAt.Time)
+	var endAt uint64
+	if e.EndAt.Valid {
+		endAt = uint64(e.EndAt.Time.Unix())
 	}
-	endAtSerialised := make([]byte, 4)
-	binary.BigEndian.PutUint32(endAtSerialised, uint32(endAt&math.MaxUint32))
-	_, err = buffer.Write(endAtSerialised)
+	endAtBytes := common.BigToHash(new(big.Int).SetUint64(endAt))
```

Comment:
Is there some type of function like `BigIntFrom(uint64` we could use here instead?

Reply:
Looks like [big.NewInt](https://golang.org/pkg/math/big/#NewInt) should do it.

Reply:
My mistake, [that was for int64](https://stackoverflow.com/questions/49754100/how-to-convert-a-uint64-to-big-int-in-golang).